api/v1alpha1/xlinecluster_types.go

```diff
@@ -66,6 +116,129 @@ type XlineClusterSpec struct {
+	}
+	if s.BootstrapArgs.ClientInitialRetryTimeout != nil {
+		args = append(args, "--client-initial-retry-timeout", *s.BootstrapArgs.ClientInitialRetryTimeout)
+	}
+	if s.BootstrapArgs.ClientMaxRetryTimeout != nil {
+		args = append(args, "--client-max-retry-timeout", *s.BootstrapArgs.ClientMaxRetryTimeout)
+	}
+	if s.BootstrapArgs.GcInterval != nil {
+		args = append(args, "--gc-interval", *s.BootstrapArgs.GcInterval)
+	}
+	if s.BootstrapArgs.RangeRetryTimeout != nil {
+		args = append(args, "--range-retry-timeout", *s.BootstrapArgs.RangeRetryTimeout)
+	}
+	if s.BootstrapArgs.CompactTimeout != nil {
+		args = append(args, "--compact-timeout", *s.BootstrapArgs.CompactTimeout)
+	}
+	if s.BootstrapArgs.SyncVictimsInterval != nil {
+		args = append(args, "--sync-victims-interval", *s.BootstrapArgs.SyncVictimsInterval)
+	}
+	if s.BootstrapArgs.WatchProgressNotifyInterval != nil {
+		args = append(args, "--watch-progress-notify-interval", *s.BootstrapArgs.WatchProgressNotifyInterval)
+	}
+	if s.BootstrapArgs.CurpDir != nil {
+		args = append(args, "--curp-dir", *s.BootstrapArgs.CurpDir)
+	}
+	if s.BootstrapArgs.CompactSleepInterval != nil {
+		args = append(args, "--compact-sleep-interval", *s.BootstrapArgs.CompactSleepInterval)
+	}
+	if s.BootstrapArgs.AutoCompactMode != nil {
+		args = append(args, "--auto-compact-mode", *s.BootstrapArgs.AutoCompactMode)
+	}
+	if s.BootstrapArgs.AutoPeriodicRetention != nil {
+		args = append(args, "--auto-periodic-retention", *s.BootstrapArgs.AutoPeriodicRetention)
+	}
+	if s.BootstrapArgs.AutoRevisionRetention != nil {
+		args = append(args, "--auto-revision-retention", *s.BootstrapArgs.AutoRevisionRetention)
+	}
+	if s.BootstrapArgs.InitialClusterState != nil {
+		args = append(args, "--initial-cluster-state", *s.BootstrapArgs.InitialClusterState)
+	}
+	if s.BootstrapArgs.RetryCount != nil {
+		args = append(args, "--retry-count", strconv.Itoa(*s.BootstrapArgs.RetryCount))
+	}
+	if s.BootstrapArgs.BatchMaxSize != nil {
+		args = append(args, "--batch-max-size", strconv.Itoa(*s.BootstrapArgs.BatchMaxSize))
+	}
+	if s.BootstrapArgs.FollowerTimeoutTicks != nil {
+		args = append(args, "--follower-timeout-ticks", strconv.Itoa(*s.BootstrapArgs.FollowerTimeoutTicks))
+	}
+	if s.BootstrapArgs.CandidateTimeoutTicks != nil {
+		args = append(args, "--candidate-timeout-ticks", strconv.Itoa(*s.BootstrapArgs.CandidateTimeoutTicks))
+	}
+	if s.BootstrapArgs.LogEntriesCap != nil {
+		args = append(args, "--log-entries-cap", strconv.Itoa(*s.BootstrapArgs.LogEntriesCap))
+	}
+	if s.BootstrapArgs.CmdWorkers != nil {
+		args = append(args, "--cmd-workers", strconv.Itoa(*s.BootstrapArgs.CmdWorkers))
+	}
+	if s.BootstrapArgs.CompactBatchSize != nil {
+		args = append(args, "--compact-batch-size", strconv.Itoa(*s.BootstrapArgs.CompactBatchSize))
+	}
+	if s.BootstrapArgs.Quota != nil {
+		args = append(args, "--quota", strconv.Itoa(*s.BootstrapArgs.Quota))
+	}
+	return args
+}
```

Comment:
We can directly use `json.Marshal` to serialize the XlineArgs and then convert it to string. Any empty field will be skipped. We don't need to check every field in the XlineArgs one by one.

Reply:
json format differs from arguments list much. When parsing a JSON string, writing some fragile code, such as trimming idents and separators.

Reply:
Here is an example:
```go
```